JavaScript/JavaScript-Classes.md

```diff
@@ -0,0 +1,171 @@
+houseMethods.closeDoor = function(){
+	this.door = 'closed'; //Add method to delegate fallback object
+
+};
+
+var house = House('red'); //instantiation pattern
+
+````
+This is not as efficient as delegating through fallback.
+
+####Prototypal
+Fallbacks are a back-up plan for objects, they are the cornerstone of **prototypal** style.  If a method or property is called on an object, where it does not exist for that object, JavaScript will check if it's defined on its fallback object. (See the README on [Inheritance and JavaScript](https://github.com/codingforeveryone/READMEs/blob/master/JavaScript/inheritance-and-javascript.md)).  Every function has a property called prototype where the fallback methods are stored.  So in the **prototypal** House function, **the House function's property (House.prototype) is explicitly delegated as the fallback location for EVERY house object created by House**.  So every method in House.prototype is available to every object created by House.  The advantage is that methods are not duplicated in memory. However, it utilises more code than the others.
+
+````
+var House = function(colour){
+	var obj = Object.create(House.prototype); // generate object
+	
+	obj.colour = colour;
+	obj.door = 'open'; //assign properties
+	
+	return obj;
+};
+
+//(Automatically generated by interpreter)
+//House.prototype = {};
+
+House.prototype.openDoor = function(){
+	this.door = 'open';
+};
+
+House.prototype.closeDoor = function(){
+	this.door = 'closed';
+};
+
+var house = House('red'); //instantiation pattern
+
+````
+
+####Pseudoclassical
+
+Lastly we come to **pseudoclassical**.  Instead of assigning the Object.create(House.prototype) to a new variable, it is assigned to `this` for the purpose of simple property assignment and method creation.  **The interpreter will do this automatically 'under the bonnet', and return the object, so long as the keyword `new` initialised at instantiation**  `var house = new House('red')`.  The disadvantage is that object creation is not that clear.
+
+````
+var House = function(colour){
+	//(Automatically generated by interpreter)
+	//var this = Object.create(House.prototype);
+	this.colour = colour;
+	this.door = 'open';
+
+	//(Automatically generated by interpreter)
+	//return this;
+};
+
+//(Automatically generated by interpreter)
+//House.property = {}
+
+House.prototype.openDoor = function(){
+	this.door = 'open';
+};
+
+House.prototype.closeDoor = function(){
+	this.door = 'closed';
+};
+
+var house = new House('red'); //instantiation pattern
+
+
+````
+
+Instantiation happens when a functional class is utilized to create a new object,
+ 
+`var house = House('red')` 
+     
+Note that Pseudoclassical is the only style that makes use of the `new` keyword, the other three classes are used like a function call.
+
+####Summary
+None of the above are better or worse to use, they are just styles, and it is up to you as the programmer to decide which your program would benefit the most from.
```

Comment:
This summary would be a great place to briefly restate the case for each style. Or just say 'For example if you are working with very large numbers of examples, you might choose function-shared rather than functional.'